cmd/strelaysrv/listener.go

```diff
@@ -149,7 +149,10 @@ func protocolConnectionHandler(tcpConn net.Conn, config *tls.Config) {
 				protocol.WriteMessage(conn, protocol.ResponseSuccess)
 
 			case protocol.ConnectRequest:
-				requestedPeer := syncthingprotocol.DeviceIDFromBytes(msg.ID)
+				requestedPeer, err := syncthingprotocol.DeviceIDFromBytes(msg.ID)
+				if err != nil {
+					panic(err)
```

Comment:
I am not sure we want to take the relay down because someone sent a malicious message.

Reply:
Neither am I. In this case it just mirrors the status quo, where `DeviceIDFromBytes()` itself would `panic()`.

What do you suggest we do instead?  I don't know the relay server code at all, so didn't see an obvious better alternative.

Reply:
If possible, drop the connection and continue whatever the outer loop is / return from the handler (I didn't look closely).

Reply:
I've noticed there are a few cases where we moved the panic out which we might have to reconsider.

Reply:
> If possible, drop the connection and continue whatever the outer loop is / return from the handler (I didn't look closely).

I'm not quite sure I understand the Go spec correctly.  When panicking inside a goroutine, will the whole program crash or just that specific routine?  If it is the latter, then the `panic()` call is appropriate, aborting the current request and continuing the connection accept loop.

Reply:
No, the program crashes, unless specific circumstances (which don't apply here).

Reply:
Panic brings down everything, not just a goroutine, and shouldn't be used for control flow (like e.g. python exceptions). Closing the connection and aborting (return, break, ) on `err != nil` is the simple and correct way to go here.

Reply:
Something along these lines: 1fb2d81a318a840446c2ca3c0e754883e66b2b5d?